Source/Core/Duality/Resources/RectAtlas.cs

```diff
@@ -0,0 +1,497 @@
+		/// </summary>
+		public Rect this[int index]
+		{
+			get { return this.items[index].Rect; }
+			set { this.items.Data[index].Rect = value; }
+		}
+
+		/// <summary>
+		/// [GET] Gets the first rect with the given tag
+		/// </summary>
+		public Rect this[string tag]
+		{
+			get { return this.items[this.tags[tag][0]].Rect; }
+		}
+
+		public RectAtlas()
+		{
+			this.items = new RawList<RectAtlasItem>();
+		}
+
+		public RectAtlas(int count)
+		{
+			this.items = new RawList<RectAtlasItem>(count);
+		}
+
+		public RectAtlas(IEnumerable<Rect> rects)
+		{
+			if (rects == null)
+				throw new ArgumentNullException("rects");
+			this.items = new RawList<RectAtlasItem>();
+			foreach (var r in rects)
+			{
+				this.items.Add(new RectAtlasItem
+				{
+					Rect = r
+				});
+			}
+		}
+
+		/// <summary>
+		/// Creates a copy of the given <see cref="RectAtlas"/>.
+		/// </summary>
+		/// <exception cref="ArgumentNullException"><paramref name="other"/> was null.</exception>
+		public RectAtlas(RectAtlas other)
+		{
+			if (other == null)
+				throw new ArgumentNullException("other");
+			this.items = new RawList<RectAtlasItem>(other.items);
+			if (other.tags != null)
+			{
+				this.tags = new Dictionary<string, List<int>>();
+				foreach (var kvp in other.tags)
+				{
+					this.tags.Add(kvp.Key, kvp.Value.ToList());
+				}
+			}
+		}
+
+		/// <summary>
+		/// Adds the given rect to the atlas.
+		/// </summary>
+		/// <param name="item">The rect to add</param>
+		public void Add(Rect item)
+		{
+			this.items.Add(new RectAtlasItem{ Rect = item });
+		}
+
+		/// <summary>
+		/// Clears the atlas of all rects and associated information
+		/// </summary>
+		public void Clear()
+		{
+			this.items.Clear();
+			this.tags = null;
+		}
+
+		/// <summary>
+		/// Determines whether or not this atlas contains the given rect
+		/// </summary>
+		/// <param name="item">The atlas to look for</param>
+		/// <returns>Whether or not the rect is contained in the atlas</returns>
+		public bool Contains(Rect item)
+		{
+			for (int i = 0; i < this.items.Count; i++)
+			{
+				if (this.items[i].Rect == item)
+					return true;
+			}
+			return false;
+		}
+
+		/// <summary>
+		/// Copies the rects in the atlas to the given array starting at the given index.
+		/// </summary>
+		/// <param name="array">The array to copy the atlas rects to</param>
+		/// <param name="arrayIndex">The index within the given array to place the atlas rects</param>
+		public void CopyTo(Rect[] array, int arrayIndex)
+		{
+			for (int i = 0; i < this.items.Count; i++)
+			{
+				array[i + arrayIndex] = this.items[i].Rect;
+			}
+		}
+
+		/// <summary>
+		/// Removes the given rect from the atlas and returns whether
+		/// or not the rect was successfully removed.
+		/// </summary>
+		/// <param name="item">The rect to remove</param>
+		/// <returns>Whether or not the given rect was removed from the atlas</returns>
+		public bool Remove(Rect item)
```

Comment:
`IList<T>` legacy: Removal by value comparison makes little sense. Instead, by-index and by-tag should be preferred.